src/libcore/ptr/mod.rs

```diff
@@ -1286,7 +1286,22 @@ impl<T: ?Sized> *const T {
     /// }
     /// ```
     #[unstable(feature = "ptr_offset_from", issue = "41079")]
+    #[cfg(not(bootstrap))]
+    #[rustc_const_unstable(feature = "const_ptr_offset_from")]
     #[inline]
+    pub const unsafe fn offset_from(self, origin: *const T) -> isize where T: Sized {
+        let pointee_size = mem::size_of::<T>();
+        let ok = 0 < pointee_size && pointee_size <= isize::max_value() as usize;
+        // assert that the pointee size is valid in a const eval compatible way
+        // FIXME: do this with a real assert at some point
+        [()][(!ok) as usize];
```

Comment:
This is odd, was this present in earlier rounds of review? And given that this is an unstable const fn, why is it needed?

Reply:
Even unstable const fn can't do `if`

It may not have been there in earlier versions, I refactored this a few times, but I couldn't figure out how to make an intrinsic panic properly.

Reply:
I thought we have `internal_unstable` for `const fn` these days to locally opt-in to unstable features?

Reply:
It's not an unstable feature. There's no way to do `if` in constants or const fn without -Zunleash-the-miri-inside-of-you which is unsound and useless outside tests

Reply:
Oh right, makes sense.